streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java

```diff
@@ -154,4 +155,51 @@ public static void purgeLocalStreamsState(Properties streamsConfiguration) throw
         produceKeyValuesSynchronously(topic, keyedRecords, producerConfig);
     }
 
+    /**
+     * Wait until enough data (key-value records) has been consumed
+     * @param consumerConfig Kafka Consumer configuration
+     * @param topic          Topic to consume grom
+     * @param expectedNumRecords Minimum number of expected records
+     * @param waitTime       Upper bound in waiting time
+     * @return All the records consumed, or null if no records are consumed
+     * @throws InterruptedException
+     */
+    public static <K, V> List<KeyValue<K, V>> waitUntilKeyValuesMatch(Properties consumerConfig, String topic,
+                                                                     int expectedNumRecords,
+                                                                     long waitTime) throws InterruptedException {
```

Comment:
If we have a common default, it may make sense to have an overloaded method without `waitTime`.